report_helper.php

```diff
@@ -2667,15 +2671,21 @@ private static function reportGridGetReloadUrl($sortAndPageUrlParams) {
   private static function getReportGridParametersForm($response, $options, $params) {
     if ($options['autoParamsForm'] || $options['paramsOnly']) {
       $r = '';
+      // The building of params form has been moved here (earlier in the function than previously) as we
+      // need any changes to $hasVisibleContent earlier in the function than before.
+      $builtParamsForm = self::build_params_form(array_merge($options, array('form'=>$response['parameterRequest'], 'defaults'=>$params)), $hasVisibleContent);
       // The form must use POST, because polygon parameters can be too large for GET.
       if (isset($options['completeParamsForm']) && $options['completeParamsForm']) {
         $cls = $options['paramsInMapToolbar'] ? 'no-border' : '';
         if (!empty($options['fieldsetClass']))
           $cls .= ' '.$options['fieldsetClass'];
         $r .= '<form action="'.$_SERVER['REQUEST_URI'].'" method="post" id="'.$options['reportGroup'].'-params">'."\n<fieldset class=\"$cls\">";
-        if (!$options['paramsInMapToolbar'])
+        // Don't use the fieldset legend in toolbar mode,
+        // and also only if the params have visible content.
+        if (!$options['paramsInMapToolbar'] && $hasVisibleContent == true) {
```

Comment:
Bools in caps and === as no need for type conversion.  So could be `$hasVisibleContent === TRUE`, or you can just write `$hasVisibleContent`